services/src/contexts/migrations/migration_0004_dataset_listing_provider_prio.rs

```diff
@@ -0,0 +1,141 @@
+use async_trait::async_trait;
+use tokio_postgres::Transaction;
+
+use crate::error::Result;
+
+use super::database_migration::{DatabaseVersion, Migration};
+
+/// This migration adds the dataset layer listing provider prios and moves the description column to the definition types
+pub struct Migration0004DatasetListingProviderPrio;
+
+#[async_trait]
+impl Migration for Migration0004DatasetListingProviderPrio {
+    fn prev_version(&self) -> Option<DatabaseVersion> {
+        Some("0003_gbif_config".into())
+    }
+
+    fn version(&self) -> DatabaseVersion {
+        "0004_dataset_listing_provider_prio".into()
+    }
+
+    async fn migrate(&self, tx: &Transaction<'_>) -> Result<()> {
+        // add priority column to layer_providers table and move description column to the definition types
+        tx.batch_execute(
+            r#"
+            ALTER TABLE layer_providers
+            ADD COLUMN priority smallint NOT NULL DEFAULT 0;
+
+            ALTER TYPE "DatasetLayerListingProviderDefinition"
+            ADD ATTRIBUTE priority smallint;
+
+            ALTER TYPE "ArunaDataProviderDefinition"
+            ADD ATTRIBUTE description text;
+            ALTER TYPE "ArunaDataProviderDefinition"
+            ADD ATTRIBUTE priority smallint;
```

Comment:
this is already stored in the layer_providers table. We could remove it from the corresponding Db types and not store it in the database.